### ISSUE TITLE
interp: add dotglob support

### ISSUE THREAD

**Issue by aucampia** [mvdan/sh#1022]
Bash supports dotglob, but as far as I can tell `mvdan.cc/sh/v3/cmd/gosh` does not:

```
$ go run mvdan.cc/sh/v3/cmd/gosh@latest -c 'shopt -s globstar; shopt -s dotglob; printf "%s\n" ./**/README*'
shopt: invalid option name "dotglob" "off" ("on" not supported)
```

Just want to know if you are open to including support for it.

**Comment by mvdan** [comment]
Yes of course - within reason, any useful Bash feature should be added to the interpreter.

**Comment by mvdan** [comment]
@aucampia this is now done at master, if you want to give it a go. cc @andreynering since I see a mention of a Taskfile PR above.
